packages/discount-code-generator/src/main.js

```diff
@@ -1,3 +1,62 @@
-export default function findSum (a, b) {
-  return a + b
+/* @flow */
+import TokenGenerator from 'tokgen'
+
+import type {
+  CodeData,
+  CodeDataArray,
+  CodeOptions,
+} from 'types/sdk'
+
+// The discountCodeGenerator function takes 2 arguments {options} and {data}
+// The {options} will specify the semantics and quantity of discount codes
+// Will throw if no quantity `number` is specified
+// Defaults to 11-digit codes if no length `number` is specified
+// Can also specify an optional prefix `string`
+// options = {
+//   quantity: 100,
+//   length: 15,
+//   prefix: CT,
+// }
+//
+// The {data} should have the attributes of the discount codes
+// data = {
+//   name: {
+//    en: 'foo'
+//   },
+//   description: {
+//    en: 'bar'
+//   },
+//   cartDiscounts: [],
+//   cartPredicate: 'some predicate',
+//   isActive: true,
+//   maxApplicationsPerCustomer: 10,
+//   maxApplicationsPerCustomer: 2,
+//  }
+
+export default function discountCodeGenerator (
+  options: CodeOptions,
```

Comment:
Params are currently in one `options` object. The `data` object is the one being mutated